--- ./config/libc/glibc.in.orig	2011-10-25 14:43:11.000000000 +0200
+++ ./config/libc/glibc.in	2011-10-25 14:43:52.000000000 +0200
@@ -41,6 +41,11 @@
 # Don't remove next line
 # CT_INSERT_VERSION_BELOW
 
+config LIBC_GLIBC_V_2_14_1
+    bool
+    prompt "2.14.1 (EXPERIMENTAL)"
+    depends on EXPERIMENTAL
+
 config LIBC_GLIBC_V_2_14
     bool
     prompt "2.14 (EXPERIMENTAL)"
@@ -118,6 +123,7 @@
     string
 # Don't remove next line
 # CT_INSERT_VERSION_STRING_BELOW
+    default "2.14.1" if LIBC_GLIBC_V_2_14_1
     default "2.14" if LIBC_GLIBC_V_2_14
     default "2.13" if LIBC_GLIBC_V_2_13
     default "2.12.2" if LIBC_GLIBC_V_2_12_2
