diff -r 8bcae97e8fd5 config/arch/sparc.in
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ ct-ng-1.7.2/config/arch/sparc.in	Wed Dec 29 21:49:45 2010 +0100
@@ -0,0 +1,12 @@
+# MIPS specific config options
+
+config ARCH_sparc
+    select ARCH_SUPPORTS_32
+    select ARCH_SUPPORTS_64
+    select ARCH_DEFAULT_32
+    select ARCH_USE_MMU
+    select ARCH_SUPPORT_CPU
+    help
+      The SUN SPARC architecture, as defined by:
+        32 bit: http://www.sparc.org/standards/V8.pdf
+        64 bit: http://www.sparc.org/standards/SPARCV9.pdf
diff -r 8bcae97e8fd5 scripts/build/arch/sparc.sh
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ ct-ng-1.7.2/scripts/build/arch/sparc.sh	Wed Dec 29 21:49:45 2010 +0100
@@ -0,0 +1,10 @@
+# Compute sparc-specific values
+CT_DoArchTupleValues() {
+    # That's the only thing to override
+    if [ "${CT_ARCH_64}" = "y" ]; then
+        CT_TARGET_ARCH="${CT_ARCH}64"
+    else
+        CT_TARGET_ARCH="${CT_ARCH}"
+    fi
+
+}
