--- ./config/cc/gcc.in.orig	2013-10-02 21:57:17.738496339 +0200
+++ ./config/cc/gcc.in	2013-10-02 21:59:04.588729753 +0200
@@ -38,7 +38,7 @@
 
 config CC_V_linaro_4_7
     bool
-    prompt "linaro-4.7-2013.06"
+    prompt "linaro-4.7-2013.09"
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_7
 
@@ -386,7 +386,7 @@
     string
 # Don't remove next line
 # CT_INSERT_VERSION_STRING_BELOW
-    default "linaro-4.7-2013.06" if CC_V_linaro_4_7
+    default "linaro-4.7-2013.09" if CC_V_linaro_4_7
     default "4.7.3" if CC_V_4_7_3
     default "4.7.2" if CC_V_4_7_2
     default "4.7.1" if CC_V_4_7_1
