diff -urN crosstool-ng-1.19.0-orig/config/cc/gcc.in crosstool-ng-1.19.0/config/cc/gcc.in
--- crosstool-ng-1.19.0-orig/config/cc/gcc.in	2013-10-02 20:06:22.000000000 +0200
+++ crosstool-ng-1.19.0/config/cc/gcc.in	2014-08-01 22:49:07.166829113 +0200
@@ -54,7 +54,7 @@
 
 config CC_V_linaro_4_7
     bool
-    prompt "linaro-4.7-2013.06-1"
+    prompt "linaro-4.7-2013.11"
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_7
 
@@ -446,7 +446,7 @@
     default "linaro-4.8-2013.06-1" if CC_V_linaro_4_8
     default "4.8.1" if CC_V_4_8_1
     default "4.8.0" if CC_V_4_8_0
-    default "linaro-4.7-2013.06-1" if CC_V_linaro_4_7
+    default "linaro-4.7-2013.11" if CC_V_linaro_4_7
     default "4.7.3" if CC_V_4_7_3
     default "4.7.2" if CC_V_4_7_2
     default "4.7.1" if CC_V_4_7_1
