--- ./config/libc/glibc.in.orig	2013-12-24 12:11:30.135911010 +0100
+++ ./config/libc/glibc.in	2013-12-24 12:11:48.651910762 +0100
@@ -14,6 +14,10 @@
 # Don't remove next line
 # CT_INSERT_VERSION_BELOW
 
+config LIBC_GLIBC_V_2_18
+    bool
+    prompt "2.18"
+
 config LIBC_GLIBC_V_2_17
     bool
     prompt "2.17"
@@ -73,6 +77,7 @@
     string
 # Don't remove next line
 # CT_INSERT_VERSION_STRING_BELOW
+    default "2.18" if LIBC_GLIBC_V_2_18
     default "2.17" if LIBC_GLIBC_V_2_17
     default "2.16.0" if LIBC_GLIBC_V_2_16_0
     default "2.15" if LIBC_GLIBC_V_2_15
