--- ./configure.orig	2011-11-08 19:08:29.801218761 +0100
+++ ./configure	2011-11-08 19:08:44.503218761 +0100
@@ -359,12 +359,10 @@
              err="'libtool' 1.5.26 or above was not found"
 has_or_abort prog=stat
 has_or_abort prog="aria2c curl wget"
-has_or_abort prog=cvs
 has_or_abort prog=patch
 has_or_abort prog=tar
 has_or_abort prog=gzip
 has_or_abort prog=bzip2
-has_or_abort prog=lzma
 has_or_abort prog=readlink
 has_or_abort prog=objcopy var=objcopy
 has_or_abort prog=objdump var=objdump
