--- crosstool-ng-1.7.2/scripts/build/cc/gcc.sh	2011-08-17 00:02:07 +0200
+++ ct-ng-1.7.2/scripts/build/cc/gcc.sh	2011-08-19 14:19:51 +0200
@@ -154,7 +154,7 @@
     # When companion libraries are build static (eg !shared),
     # the libstdc++ is not pulled automatically, although it
     # is needed. Shoe-horn it in our LDFLAGS
-    if [ "${CT_COMPLIBS_SHARED}" != "y" ]; then
+    if [ "${CT_CC_LANG_CXX}" = "y" -a "${CT_COMPLIBS_SHARED}" != "y" ]; then
         core_LDFLAGS='-lstdc++'
     fi
     if [ "${CT_CC_GCC_USE_GMP_MPFR}" = "y" ]; then
@@ -329,7 +329,7 @@
     # When companion libraries are build static (eg !shared),
     # the libstdc++ is not pulled automatically, although it
     # is needed. Shoe-horn it in our LDFLAGS
-    if [ "${CT_COMPLIBS_SHARED}" != "y" ]; then
+    if [ "${CT_CC_LANG_CXX}" = "y" -a "${CT_COMPLIBS_SHARED}" != "y" ]; then
         final_LDFLAGS='-lstdc++'
     fi
     if [ "${CT_CC_GCC_USE_GMP_MPFR}" = "y" ]; then
